### PR TITLE
Prevent option change event firing when changes occur in component

### PR DESCRIPTION
**Pull request by anna-zhernovkova**
Fix T614207

### CODE REVIEW

File: tests/src/core/component.spec.ts [hunk 1]
```diff
@@ -194,6 +194,23 @@ describe('DevExtreme Angular widget', () => {
         expect(testSpy).toHaveBeenCalledTimes(1);
     }));
 
+    it('should not emit testOptionChange event when changes happen in component (T614207)', async(() => {
```
Comment (ovchinnikov) [inline tests/src/core/component.spec.ts]: Is there any reason to make this test asynchronous?

Comment (anna-zhernovkova) [reply tests/src/core/component.spec.ts]: No, I will fix it

File: src/core/component.ts [hunk 2]
```diff
@@ -93,6 +93,15 @@ export abstract class DxComponent implements OnChanges, OnInit, DoCheck, AfterCo
             this.instance.endUpdate();
         }
     }
+    lockEventFire(name: string) {
+        this.eventHelper.lockEventFire.push(name + 'Change');
```
Comment (ovchinnikov) [inline src/core/component.ts]: Is it necessary to have an array here? It seems that it's impossible to have several locked events simultaneously

Comment (anna-zhernovkova) [reply src/core/component.ts]: I found the case, where we have two events in the array, but it's checked in the correct order. So, we can get rid of array here.

**Pull request merged**